packages/flutter/lib/src/rendering/flow.dart

```diff
@@ -0,0 +1,345 @@
+  /// be adjusted to be as close to the returned size as possible while still
+  /// respecting the constraints.
+  ///
+  /// If this function depends on information other than the given constraints,
+  /// override [shouldRelayout] to indicate when when the container should
+  /// relayout.
+  Size getSize(BoxConstraints constraints) => constraints.biggest;
+
+  /// Override to control the layout constraints given to each child.
+  ///
+  /// By default, the children will receive the given constraints, which are the
+  /// constraints the constraints used to size the container. The children need
+  /// not respect the given constraints, but they are required to respect the
+  /// returned constraints.
+  ///
+  /// If this function depends on information other than the given constraints,
+  /// override [shouldRelayout] to indicate when when the container should
+  /// relayout.
+  BoxConstraints getConstraintsForChild(int i, BoxConstraints constraints) => constraints;
+
+  /// Override to paint the children of the flow.
+  ///
+  /// Children can be painted in any order, but each child can be painted at
+  /// most once. Although the container clips the children to its own bounds, it
+  /// is more efficient to skip painting a child altogether rather than having
+  /// it paint entirely outside the container's clip.
+  ///
+  /// To paint a child, call [FlowPaintingContext.paintChild] on the given
+  /// [context]. The given context is valid only within the scope of this
+  /// function call and contains information (such as the size of the container)
+  /// that is useful for picking transformation matrices for the children.
+  void paintChildren(FlowPaintingContext context);
+
+  /// Override this method to return true when the children need to be laid out.
+  /// This should compare the fields of the current delegate and the given
+  /// oldDelegate and return true if the fields are such that the layout would
+  /// be different.
+  bool shouldRelayout(FlowDelegate oldDelegate) => false;
+
+  /// Override this method to return true when the children need to be
+  /// repainted. This should compare the fields of the current delegate and the
+  /// given oldDelegate and return true if the fields are such that
+  /// paintChildren would act differently.
+  ///
+  /// The delegate can also trigger a repaint if the delegate provides the
+  /// repaint animation argument to this object's constructor and that animation
+  /// ticks. Triggering a repaint using this animation-based mechanism more
+  /// efficient than rebuilding the [Flow] widget to change its delegate.
+  bool shouldRepaint(FlowDelegate oldDelegate);
+
+  /// Override this method to include additional information in the
+  /// debugging data printed by [debugDumpRenderTree] and friends.
+  ///
+  /// By default, returns the [runtimeType] of the class.
+  @override
+  String toString() => '$runtimeType';
+}
+
+class _FlowParentData extends ContainerBoxParentDataMixin<RenderBox> {
```

Comment:
this shouldn't be private, since you can get to it easily.

When documenting it once it's public, mention that "offset" is also zero for children of a RenderFlow.